src/dms/core/EdFi.DataManagementService.Core/Middleware/ResourceActionAuthorizationMiddleware.cs

```diff
@@ -23,162 +23,362 @@ internal class ResourceActionAuthorizationMiddleware(
     ILogger _logger
 ) : IPipelineStep
 {
-    public async Task Execute(RequestData requestData, Func<Task> next)
+    private static readonly Dictionary<RequestMethod, string> _methodToActionNameMapping = new()
+    {
+        { RequestMethod.POST, "Create" },
+        { RequestMethod.GET, "Read" },
+        { RequestMethod.PUT, "Update" },
+        { RequestMethod.DELETE, "Delete" },
+    };
+
+    /// <summary>
+    /// Executes the authorization middleware to validate resource access permissions.
+    /// </summary>
+    public async Task Execute(RequestInfo requestInfo, Func<Task> next)
```

Comment:
I refactored this class, breaking it up into methods and replacing the ActionResolver class that was only used here as a method -> action name mapping.